MSAL/src/native_auth/controllers/MSALNativeAuthBaseController.swift

```diff
@@ -163,4 +163,11 @@ class MSALNativeAuthBaseController {
             }
         }
     }
+
+    func updateContext(_ previousContext: MSIDRequestContext, with correlationId: UUID?) -> MSALNativeAuthRequestContext {
+        return MSALNativeAuthRequestContext(
+            correlationId: correlationId ?? previousContext.correlationId(),
```

Comment:
If I understand correctly, don't we expect the new correlationId to be the same as the old one in most cases? Would it be useful/interesting to log this as a warning when they are different, or even just at info level?

Reply:
Yes, good suggestion 👍. I've added it as a warning. If this happens it means that something is wrong in the backend.